feature-libs/product/configurators/rulebased/root/interceptor/cpq-access-storage.service.ts

```diff
@@ -0,0 +1,67 @@
+import { Inject, Injectable, InjectionToken } from '@angular/core';
+import { Observable, timer } from 'rxjs';
+import { expand, filter, shareReplay, switchMap } from 'rxjs/operators';
+import { CpqAccessData } from './cpq-access-data.models';
+import { CpqAccessLoaderService } from './cpq-access-loader.service';
+
+const ONE_DAY = 24 * 60 * 60 * 1000;
+
+@Injectable({ providedIn: 'root' })
+export class CpqAccessStorageService {
+  /* We should stop using/sending a token shortly before expiration,
+   * to avoid that it is actaully expired when evaluated in the target system */
+  static TOKEN_EXPIRATION_BUFFER = new InjectionToken<number>(
+    'TOKEN_EXPIRATION_BUFFER',
+    {
+      factory: CpqAccessStorageService.defaultValue,
+    }
+  );
+
+  constructor(
+    protected cpqAccessLoaderService: CpqAccessLoaderService,
+    @Inject(CpqAccessStorageService.TOKEN_EXPIRATION_BUFFER)
+    protected EXPIRATION_BUFFER: number
+  ) {}
+
+  protected cpqAccessData: Observable<CpqAccessData>;
+
+  private static defaultValue() {
```

Comment:
Please never use 'private', and have this as default configuration (not publicly exposed. See e.g. /feature-libs/product/configurators/rulebased/components/config/default-message-config.ts.
Customers then can easily override